shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -877,13 +899,31 @@ SemanticsObject hitTest(float[] point) {
         boolean isFocusable() {
             int scrollableActions = Action.SCROLL_RIGHT.value | Action.SCROLL_LEFT.value
                     | Action.SCROLL_UP.value | Action.SCROLL_DOWN.value;
+            if (hasFlag(Flag.IS_ROUTE)) {
```

Comment:
Do we enforce (on the framework side?) that a SemanticsNode with the route flag never has any other relevant information set? For example, can we ensure that the following code doesn't drop semantic information on the floor?

```dart
new Semantics(
  route: true,
  value: 'my route',
  child: new Semantics(
    button: true,
    value: 'click me',
    child: new Placeholder(),
  ),
);
```

We might have to ensure that semantics information never get implicitly merged into a node with route: true.

Reply:
I think we may actually want the opposite — a way to just stick "route:true" on an otherwise pre-existing semantic node, so that the text that's on the screen already gets used as the route name.

Reply:
@Hixie does it make sense to handle this in the accessibility bridge, or does this need to be added to the Semantics logic on the framework side?

Reply:
Spoke offline with @yjbanov and @goderbauer on this - we really don't have a good way of merging the semantics to guarantee that we get a reasonable default here.  Instead, I can change the framework side so that `route: true` always sets `explicitChildNodes` so that the semantics are not incorrectly merged.

Reply:
That won't work, since we would no longer be able to do something like wrap an app bar in a semantics with route:true and autmatically get a name due to merging.